### ISSUE TITLE
Stacking error handlers

### ISSUE THREAD

**Issue by derrabus** [vimeo/psalm#10646]
I'd like to register a temporary error handler that decorates my global error handler. I usually do that this way:

```php
$previous = null;
$previous = set_error_handler(static function () use (&$previous): bool {
    // Special error handling here.
    
    // Fall back to the global error handler
    return is_callable($previous) && $previous(...func_get_args());
});

try {
    // Do something
} finally {
    restore_error_handler();
}
```

If you know a more elegant way to do this, please enlighten me. Psalm however does not like at all what I'm doing here. 🙈 

Minimal version:

```php
<?php

$previous = null;
$previous = set_error_handler(static function () use (&$previous): bool {
    return is_callable($previous) && $previous(...func_get_args());
});
```

https://psalm.dev/r/8d2ca309f8

> ERROR: [TypeDoesNotContainType](https://psalm.dev/056) - 5:12 - Type null for $previous is never callable
>
> ERROR: [InvalidFunctionCall](https://psalm.dev/064) - 5:38 - Cannot treat type never as callable
>
> INFO: [UnusedVariable](https://psalm.dev/077) - 4:1 - $previous is never referenced or the value is not used

All three errors are false positives imo.

**Comment by weirdan** [comment]
Psalm just needs a little help: https://psalm.dev/r/df9f6e0be7

**Comment by derrabus** [comment]
Thanks! This workaround is helpful, but then again: should Psalm need this kind of help? Shall we keep the issue open?

**Comment by weirdan** [comment]
References are intrinsically problematic to analyze (see https://en.wikipedia.org/wiki/Action_at_a_distance_(computer_programming)) , so I would say yes.

You can rewrite this as a class where (hidden) references are (implicitly) typed, so providing this additional information Psalm needs would feel more natural. E.g. something like this: https://psalm.dev/r/d5103d86ae
